### PR TITLE
Enable linux_job.yml to have dynamic secrets forwarded to user environment

### PR DESCRIPTION
**Pull request by jeanschmidt**
Export secrets as environment variables for linux reusable workflows with the formatting: `re.sub(r'[^a-zA-Z0-9_]', '', f'SECRET_{x.upper()}'.replace('-', '_'))`

### CODE REVIEW

File: .github/workflows/linux_job.yml [hunk 1]
```diff
@@ -141,42 +144,91 @@ jobs:
...
+              -v "{ os.environ.get('GITHUB_WORKSPACE', '') }/{ os.environ.get('REPOSITORY', '') }:/work" \
+              -v "{ os.environ.get('GITHUB_WORKSPACE', '') }/test-infra:/test-infra" \
+              -v "{ os.environ.get('RUNNER_ARTIFACT_DIR', '') }:/artifacts" \
+              -v "{ os.environ.get('RUNNER_TEST_RESULTS_DIR', '') }:/test-results" \
+              -v "{ os.environ.get('RUNNER_TEMP', '') }/exec_script:/exec" \
+              -v "{ os.environ.get('GITHUB_STEP_SUMMARY', '') }":"{ os.environ.get('GITHUB_STEP_SUMMARY', '') }" \
+              -w /work \
+              "{ os.environ.get('DOCKER_IMAGE', '') }"
+          ''').replace('\n', '').strip()
+          run_cmd_or_die(f'docker exec -t {container_name} /exec')
```
Comment (DanilBaibak) [inline .github/workflows/linux_job.yml:231]: Can we move this code to a separate .py file? We can also add tests to this logic.

**Review by DanilBaibak — APPROVED** [review]
Overall LGTM, but we need to test these changes carefully.

File: .github/workflows/linux_job.yml (hunk 1 above)
Comment (jeanschmidt) [reply .github/workflows/linux_job.yml:231]: I am investigating this, to understand how reusable workflows deals with files from the current repo when called from another repo...

Comment (seemethere) [reply .github/workflows/linux_job.yml:231]: We check out the test-infra repository in this job so anything you put in there should be fair game at `${GITHUB_WORKSPACE}/test-infra`

**Comment by seemethere** [comment]
q: How does this handle more than 1 secret?

**Comment by jeanschmidt** [comment]
a: you can either not specify any secret filtering, or you can specify multiple ones space-separated

**Comment by weiwangmeta** [comment]
Could this be causing https://github.com/pytorch/vision/issues/7515 and  https://github.com/pytorch/audio/actions/runs/4733965450/jobs/8402221728?pr=3271?

**Comment by weiwangmeta** [comment]
According to @atalman , https://github.com/pytorch/test-infra/pull/3956 started to cause Audio failures (on audio side this workflow passes: https://github.com/pytorch/audio/actions/runs/4678030776/jobs/8286260647, The next one that is using run_docker_with_env_secrets.py is failing).

**Comment by jeanschmidt** [comment]
The fix for the issue is being worked on https://github.com/pytorch/test-infra/pull/4045

**Pull request merged**